shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -285,11 +285,11 @@ where
             read_records_data.truncate(RawRecord::SIZE - 2);
         }
 
-        let data_length = self.decode_data_length(
-            &read_records_data[piece_offset as usize..],
-            piece_index,
-            piece_offset,
-        )?;
+        // Discard piece data before the offset
+        let mut data = read_records_data[piece_offset as usize..].to_vec();
+        drop(read_records_data);
```

Comment:
Why re-allocating?

If you know the offset beforehand, you don't need to store all raw record chunks in it to begin with, simply add `.skip(piece_offset as usize)` above at line 281.

Reply:
In another PR now.